Adjust spacings and alignment for Divi

gabrielcaires opened the pull request:
Resolves #6856

## Proposed Changes
* Adjust spacings and alignment for Divi on Frontend
* Fix basic spacing issues on the editor (Only to make it usable). 
*  Load an editor css file only for divi.

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Install and Active the divi theme
2. Go to a course with learning mode enabled
3. Check if the spacings are similar to the expected designs. 


<img width="1447" alt="Screenshot 2023-05-15 at 15 05 57" src="https://github.com/Automattic/sensei/assets/38718/61e34540-68df-4988-aa99-5c7178ac05b4">
<img width="488" alt="Screenshot 2023-05-15 at 15 08 50" src="https://github.com/Automattic/sensei/assets/38718/79831486-af11-4e9e-8a4d-3710276ba198">

<img width="489" alt="Screenshot 2023-05-15 at 15 07 48" src="https://github.com/Automattic/sensei/assets/38718/310cbd64-6518-452e-9ae0-364fdb0453b5">

<img width="1348" alt="Screenshot 2023-05-15 at 15 10 16" src="https://github.com/Automattic/sensei/assets/38718/a36cf554-d61e-47da-a9c6-8d08573c51e5">



## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [x] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [x] Code is tested on the minimum supported PHP and WordPress versions

gabrielcaires committed:
Enable to load specific css file for divi

gabrielcaires committed:
Review layout spacing

gabrielcaires committed:
Load learning mode style only for divi

gabrielcaires committed:
Fix learning mode layout issues on divi

gabrielcaires committed:
Fix formatting issues

gabrielcaires committed:
Merge branch 'update/spacing-course-theme' into update/review-div-spacing

gabrielcaires committed:
Fix lint issues

merkushin commented:
Front end looks pretty good. Here are a few issues I noticed:

Too little space between the paragraph and the video above:
<img width="1000" alt="CleanShot 2023-05-18 at 01 21 32@2x" src="https://github.com/Automattic/sensei/assets/329356/3f4bc338-d1c9-448f-bf58-cb5746e2de0c">

No space at all between the lesson name and the module name:
<img width="456" alt="CleanShot 2023-05-18 at 01 21 16@2x" src="https://github.com/Automattic/sensei/assets/329356/0c368c70-bbf6-4b27-ba7e-cd5bddddf60b">

[By design](8mY8ui4zYK1Ylw2yIFi9Lx-fi-20021387_41852), it is 24px:
<img width="508" alt="CleanShot 2023-05-18 at 01 23 50@2x" src="https://github.com/Automattic/sensei/assets/329356/79830a10-f540-4c31-8caa-cdb490468356">

Too much space in the navigation for lessons without modules:
<img width="312" alt="CleanShot 2023-05-18 at 01 19 51@2x" src="https://github.com/Automattic/sensei/assets/329356/6191f35e-2559-41a1-8584-a4012398c81f">

Too little space on the top of the content area:
<img width="875" alt="CleanShot 2023-05-18 at 01 19 17@2x" src="https://github.com/Automattic/sensei/assets/329356/07d09138-05c5-4577-891f-204c5866895e">

merkushin committed:
Merge branch 'update/spacing-course-theme' into update/review-div-spacing

gabrielcaires committed:
Merge branch 'feature/learning-mode-improvements' into update/review-div-spacing

gabrielcaires commented:
@merkushin  I didn't have success reproducing the issues after I updated the branch.
 
<img width="923" alt="Screenshot 2023-05-22 at 11 37 47" src="https://github.com/Automattic/sensei/assets/38718/ef34164a-3a89-4242-bc62-deecec0955cd">
<img width="311" alt="Screenshot 2023-05-22 at 14 05 40" src="https://github.com/Automattic/sensei/assets/38718/d3f9be3e-6689-4cfd-acbf-385e544494de">
<img width="840" alt="Screenshot 2023-05-22 at 11 36 16" src="https://github.com/Automattic/sensei/assets/38718/ec004438-f385-4537-a218-81bbbe0cbf3e">

I tried to recreate my env to try to reproduce.

Could you please check again? I am still trying to figure out if my env is reliable.

merkushin committed:
Merge branch 'feature/learning-mode-improvements' into update/review-div-spacing

merkushin commented:
@gabrielcaires 
I still see some of those issues. Course navigation on mobile has changed partially to a better, partially to a worse.
I recorded my screen. I forgot to show the active theme there, but I double-checked, it is Divi.


https://github.com/Automattic/sensei/assets/329356/14c45a15-c496-4c25-be9d-f749997670f0

Imran92 commented on includes/3rd-party/themes/divi.php:
```diff
@@ -89,4 +89,54 @@ function sensei_fix_divi_learning_mode_video_template_excerpt() {
+ * Enqueue Course theme-specific Learning Mode styles in the admin for the Site Editor and Lesson Editor.
+ */
+function sensei_admin_load_learning_mode_style_for_divi_theme() {
+	$is_course_theme = 'divi' === wp_get_theme()->get_template();
```
Should this variable be named `$is_divi_theme` or something like that?

gabrielcaires replied:
Fixed  here 2703ebb

Imran92 commented on includes/3rd-party/themes/divi.php:
```diff
@@ -89,4 +89,54 @@ function sensei_fix_divi_learning_mode_video_template_excerpt() {
+
+	if ( $is_lesson_editor || $is_site_editor ) {
+		Sensei()->assets->enqueue(
+			'div-learning-mode',
```
Should this be `divi-learning-mode`?

gabrielcaires replied:
Fixed  here 2703ebb

Imran92 commented on includes/3rd-party/themes/divi.php:
```diff
@@ -89,4 +89,54 @@ function sensei_fix_divi_learning_mode_video_template_excerpt() {
+ */
+function sensei_load_learning_mode_style_for_divi_theme() {
+	$course_id       = Sensei_Utils::get_current_course();
+	$is_course_theme = 'divi' === wp_get_theme()->get_template();
```
Should this variable be named `$is_divi_theme` or something like that?

gabrielcaires replied:
Fixed  here 2703ebb

Imran92 commented on includes/3rd-party/themes/divi.php:
```diff
@@ -89,4 +89,54 @@ function sensei_fix_divi_learning_mode_video_template_excerpt() {
+		);
+
+		Sensei()->assets->enqueue(
+			'div-learning-mode-editor',
```
Should this be `divi-learning-mode-editor`?

gabrielcaires replied:
Fixed  here 2703ebb

Imran92 commented on includes/3rd-party/themes/divi.php:
```diff
@@ -89,4 +89,54 @@ function sensei_fix_divi_learning_mode_video_template_excerpt() {
 }
+
+/**
+ * Enqueue Course theme-specific Learning Mode styles.
```
Divi?

gabrielcaires replied:
Fixed  here 2703ebb

Imran92 commented on includes/3rd-party/themes/divi.php:
```diff
@@ -89,4 +89,54 @@ function sensei_fix_divi_learning_mode_video_template_excerpt() {
+}
+
+/**
+ * Enqueue Course theme-specific Learning Mode styles in the admin for the Site Editor and Lesson Editor.
```
Divi?

gabrielcaires replied:
Fixed  here 2703ebb

Imran92 committed:
Remove case sensitivity from matching divi theme name

Imran92 commented:
I've introduced a small fix here, otherwise it wasn't working for me locally https://github.com/Automattic/sensei/pull/6883/commits/bac70547bd090c04f457194a18833a85d5d4c66d

merkushin commented:
After Imran's fix, issues with spacing in content part on desktop were resolved.
The last issue I see is the spacing in the course navigation on mobile.

![CleanShot 2023-05-23 at 13 57 43@2x](https://github.com/Automattic/sensei/assets/329356/3b065a43-40c7-4913-9266-b67554c65c37)
![CleanShot 2023-05-23 at 13 57 35@2x](https://github.com/Automattic/sensei/assets/329356/f96426f1-82db-4bba-95a7-f5c250c48882)

gabrielcaires committed:
Fix wrong theme name

gabrielcaires committed:
Remove content padding override when is admin

gabrielcaires commented:
@merkushin As I described [here](p1684853438287789-slack-C02NWDZBL0H) due to some reason that I couldn't identify, I can not reproduce it,  so I removed the change I introduced based on my env.  [8ca3169]
Could you please test it again?

merkushin reviewed (approved):
👍 

<img width="409" alt="CleanShot 2023-05-24 at 01 10 55@2x" src="https://github.com/Automattic/sensei/assets/329356/1a8609a9-da58-4fc3-8df7-33cb1a6db8ae">

Imran92 commented:
Merging into the feature branch as it's approved 🎆
